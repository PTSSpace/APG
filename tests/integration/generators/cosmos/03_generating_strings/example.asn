Module-example DEFINITIONS AUTOMATIC TAGS ::= BEGIN

  Digits ::= INTEGER(0..9)

  Simple-choice ::= SEQUENCE {
    sleep Sleep-time,
    eat SEQUENCE (SIZE (2)) OF Food, -- select two food
    visible-string IA5String (SIZE (128)),
    numeric-string NumericString (SIZE (129)),
    zip-code SEQUENCE (SIZE (5)) OF Digits -- your 5 digit zip code
  }

  Food ::= ENUMERATED {
    carrot,
    apple,
    banana
  }

  Sleep-time ::= SEQUENCE {
    begin Time,
    end Time
  }

  Time ::= SEQUENCE {
    hour INTEGER(0..23),     -- [h]
    minute INTEGER(0..59)     -- [min]
  }

END
