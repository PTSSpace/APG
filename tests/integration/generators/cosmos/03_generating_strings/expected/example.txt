# This file was autogenerated from ASN.1 model.

TELEMETRY example Simple-choice BIG_ENDIAN ""
    APPEND_ITEM         hour                                  5 UINT            ""
        UNITS h h
    APPEND_ITEM         minute                                6 UINT            ""
        UNITS min min
    APPEND_ITEM         hour                                  5 UINT            ""
        UNITS h h
    APPEND_ITEM         minute                                6 UINT            ""
        UNITS min min
    APPEND_ARRAY_ITEM   eat                                   2 UINT          4 "select two food"
        STATE carrot                          0
        STATE apple                           1
        STATE banana                          2
    APPEND_ITEM         visible-string                     1024 STRING          ""
    APPEND_ITEM         numeric-string                     1032 STRING          ""
    APPEND_ARRAY_ITEM   zip-code                              4 UINT         20 "your 5 digit zip code"
